### PR TITLE
Create the Dossier Review Page

### LINKED ISSUE #243: Create Dossier Review Page
As an initiator or reviewer, I want to be able to review a created dossier.

[Story Points: 3]
[Priority: High]
[Risk: Medium]

This is a subtask of #15

### PR DESCRIPTION
**Pull request by kennyphan100**
This PR only involves creating the UI for the Dossier Review Page. The backend functionalities for the page are not complete.

This closes #243.

*** A demo of the page is shown below.

### CODE REVIEW

**Comment by kennyphan100** [comment]
Demo:

https://github.com/JamesPartsafas/ConcordiaCurriculumManager/assets/66841718/f5e3eac3-e6e9-4fcd-8f65-ff649f67e3df

**Review by JamesPartsafas — APPROVED** [review]
Looks good, and thanks for posting a demo video for easy reviewing. I included some comments of things that don't need to be fixed immediately, but they're documented for later. 

I suspect we may need to modify the dossier details endpoint, to include info relevant to the approval pipeline, including the group names reviewing and all the relevant discussion data. I think when @Ghaith-Chrit you make the endpoint for posting discussion comments, you can make that modification to the dossier details endpoint, it kind of makes sense to put it all together, unless you encounter issues. The group names can be gotten through the ApprovalStage objects, that are already in the DTO.

File: App/src/pages/dossier/DossierReview.tsx [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+                                <Center>
+                                    <Heading as="h2" size="xl" color="brandRed">
+                                        <Text align="center">Add Message</Text>
+                                    </Heading>
+                                </Center>
+                                <Stack mb={2}>
+                                    <Text as="span">
+                                        ***You are a member of the{" "}
+                                        <Text as="span" fontWeight={"bold"} fontSize={20}>
+                                            Gina Cody School Faculty Council
```
Comment (JamesPartsafas) [inline App/src/pages/dossier/DossierReview.tsx:214]: When backend functionality is created, this group name should always be the name of the currently reviewing group.

File: App/src/pages/dossier/DossierReview.tsx [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+
+            <Box>
+                <form>
+                    <Button
+                        style="primary"
+                        variant="outline"
+                        height="40px"
+                        width="100px"
+                        margin="20px"
+                        onClick={() => navigate(BaseRoutes.Dossiers)}
```
[the remark this answers is not in the document]
Comment (kennyphan100) [reply App/src/pages/dossier/DossierReview.tsx:123]: Sounds good, I'll put a comment in the code as a reminder.

File: App/src/pages/dossier/DossierReview.tsx [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+                                <Heading margin="auto" size="xl" color="brandRed">
+                                    Changes
+                                </Heading>
+                                <Stack mb={4}>
+                                    <Heading size={"md"} color="brandBlue">
+                                        Course Creation Requests:
+                                    </Heading>
+                                    {dossierDetails?.courseCreationRequests?.map((courseCreationRequest) => (
+                                        <Text key={courseCreationRequest.id} as="u">
+                                            <Link to={BaseRoutes.ManageableGroup}>
```
[the remark this answers is not in the document]
Comment (kennyphan100) [reply App/src/pages/dossier/DossierReview.tsx:168]: Sounds good.

File: App/src/pages/dossier/DossierReview.tsx [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+                            </Stack>
+                            <Stack marginTop={16}>
+                                <Center>
+                                    <Heading as="h2" size="xl" color="brandRed">
+                                        Discussion Board
+                                    </Heading>
+                                </Center>
+                                <Stack>
+                                    <Tabs>
+                                        <TabList>
```
[the remark this answers is not in the document]
Comment (kennyphan100) [reply App/src/pages/dossier/DossierReview.tsx:251]: Ok.

File: App/src/pages/dossier/DossierReview.tsx (hunk 1 above)
Comment (kennyphan100) [reply App/src/pages/dossier/DossierReview.tsx:214]: So does this mean that other people from other groups cannot review this Dossier at the same time?
For instance, if the currently reviewing group is Department Council, then someone else from the Gina Cody School Faculty Council cannot review this Dossier?

File: App/src/pages/dossier/DossierReview.tsx [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+                                    Dossier Review
+                                </Heading>
+                                <Heading color={"black"}>{dossierDetails?.title}</Heading>
+                                <Kbd>{dossierDetails?.id}</Kbd>
+                                <Text>{dossierDetails?.description}</Text>
+                                <Text>state: {dossierStateToString(dossierDetails)}</Text>
+                                <Text>created: {dossierDetails?.createdDate?.toString()}</Text>
+                                <Text>updated: {dossierDetails?.modifiedDate?.toString()}</Text>
+                            </Stack>
+                            <Stack direction="row" spacing={4} align="center" marginBottom={10}>
```
[the remark this answers is not in the document]
Comment (kennyphan100) [reply App/src/pages/dossier/DossierReview.tsx:135]: I'm not sure how to retrieve the role of the user. Will we need an endpoint to GetUserRoleFromGroup? Or is there some other way to get the role?
Sounds good, I'll make the changes in a future PR to correct these issues.

**Pull request merged**